csharp/src/Microsoft.ML.OnnxRuntime/NativeOnnxTensorMemory.cs

```diff
@@ -12,7 +12,12 @@
 
 namespace Microsoft.ML.OnnxRuntime
 {
-    internal class NativeOnnxTensorMemory<T> : MemoryManager<T>
+    internal interface NativeMemoryHandler : IDisposable
```

Comment:
Not sure if this makes problem: why not just passing the onnxValueHandle as IntPtr instead of passing the object as `NativeMemoryHandler`? it looks like the native value will never change since the object created